ftp/src/main/scala/akka/stream/alpakka/ftp/impl/FtpBrowserGraphStage.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.ftp
+package impl
+
+import akka.Done
+import akka.stream.stage.{ GraphStageWithMaterializedValue, OutHandler }
+import akka.stream.{ Attributes, Outlet, SourceShape }
+import scala.concurrent.{ Future, Promise }
+
+private[ftp] trait FtpBrowserGraphStage[FtpClient]
+    extends GraphStageWithMaterializedValue[SourceShape[FtpFile], Future[Done]] {
+
+  def name: String
+
+  def basePath: String
+
+  def connectionSettings: RemoteFileSettings
+
+  implicit def ftpClient: FtpClient
+
+  val ftpLike: FtpLike[FtpClient]
+
+  val shape = SourceShape(Outlet[FtpFile](s"$name.out"))
+
```

Comment:
as the blocking logic is done already in this trait I think the `IODispatcher` should be specified using `initialAttributes` here rather than when creating the flows in `FtpApi` (and possibly miss out on it)